src/options.rs

```diff
@@ -47,28 +51,63 @@ impl Options {
         }
     }
 
-    pub fn setter_enabled(&self) -> bool {
-        self.setter_enabled
+    pub fn setter_prefix(&self) -> &str {
+        &self.setter_prefix
+    }
+
+    pub fn getter_enabled(&self) -> bool {
+        self.getter_enabled
+    }
+
+    pub fn _getter_visibility(&self) -> Option<quote::Tokens> {
+        if self.getter_public {
+            Some(quote!(pub))
+        } else {
+            None
+        }
+    }
+
+    pub fn _getter_prefix(&self) -> &str {
+        &self.getter_prefix
     }
 }
 
-impl<T> From<T> for Options where
-    T: IntoIterator<Item=syn::Attribute>
+impl<'a, T> From<T> for Options where
+    T: IntoIterator<Item=&'a syn::Attribute>
 {
     fn from(attributes: T) -> Self {
         trace!("Parsing struct attributes.");
-        let mut ob = OptionsBuilder::default();
-
-        for attr in attributes {
-            ob.parse_attribute(attr);
-        }
+        let mut builder = OptionsBuilder::<StructMode>::default();
+        builder.parse_attributes(attributes);
 
-        ob.into()
+        builder.into()
     }
 }
 
+pub trait OptionsBuilderMode {
+    // associated consts are not a thing yet :-/
+    // https://github.com/rust-lang/rust/issues/29646
```

Comment:
As a sidenote: associated consts would be good to convert [this](https://github.com/colin-kiegel/rust-derive-builder/blob/feature/field-overrides/src/options.rs#L221-L227) `if let {} else if let {}` pattern into a `match` which looks nicer IMO. 

If you prefer a default impl I can change it, but it would still not allow to switch to the nicer match-syntax. :-/

PS: On a high level I am a bit worried about differentiating the attribute names for the struct and field case. On the plus side it's grammatically better, but on the minus side it blocks copy-pasting attributes between levels. I have a slight preference to change that and reduce the complexity of the API. What do you think?